controllers/pre_backup.go

```diff
@@ -259,7 +257,7 @@ func prepareImportedClusters(ctx context.Context,
 				msaAddon := &addonv1alpha1.ManagedClusterAddOn{}
```

Comment:
Is it possible to also update the install namespace of the existing managed-serviceaccount addon created by the backup operator?

Reply:
I saw there is a label in backup-operator code, line 261 - line 263:
```
				labels := map[string]string{
					msa_label: msa_service_name,
				}
```
If this can indicate that the managed-serviceaccount addon was created by DR?

Reply:
@zhujian7 the problem is that this label is kind of generic (it's just "authentication.open-cluster-management.io/is-managed-serviceaccount": "auto-import-account") so I'm not sure if it can be added by others as well.  I don't know this history of why this label was applied.

There is also still the question of whether it's safe for us to change the namespace.  Is this something that won't cause any issues?  If the namespace must be fixed, I'd prefer this is implemented & tested in the managed service account addon itself, rather than backup&restore needing to figure out all the permutations of changing on clusters  out there.  I'm not clear on whether we can simply change the installNamespace on this addon (I assume we'd reset it to "") if it's already set to something non-standard.

Reply:
Hi @tesshuflower , sorry to make you confused, let me give some background and try to answer your questions:

> There is also still the question of whether it's safe for us to change the namespace. Is this something that won't cause any issues?

I tested changing the install namespace, it will delete the agent in the old ns, and create another one in the new ns. and **the old ns (in the customer case "openshift-storage") will be deleted**. but after changing it to "open-cluster-management-agent-addon", deleting the addon will not trigger to delete the "open-cluster-management-agent-addon" ns, once [this](https://github.com/open-cluster-management-io/managed-serviceaccount/pull/74) is merged. (I can test again tomorrow if you're worried)

> If the namespace must be fixed, I'd prefer this is implemented & tested in the managed service account addon itself

For 2.7 and 2.8, it is not required to be a fixed value, it is configurable, but deleting the addon will trigger to delete the install ns, so I submit a PR to [not delete the ns if it is the default ns](https://github.com/stolostron/managed-serviceaccount/pull/36) to address the customer issue we have now. why do not we orphan any ns no matter what name it is but just the "open-cluster-management-agent-addon", because the namespace created by enabling the managed-serviceaccount addon should be deleted when disabling the addon, for example, if the install ns is set to "open-cluster-management-service-account", enabling the addon will create the ns on the managed cluster, it should not remain when deleting the addon. So we recommend changing the ns to empty/default "open-cluster-management-agent-addon".


For 2.9, because we are planning to change it to a template-type addon, unfortunately, there are some issues we can not address now, as a result, we want to only support a fixed value in 2.9, but in the future, it should be configurable. And I think it is still not decided to change it to a fixed value before Train 6 starts, I am collecting feedback from managed serviceaccount users, if you strongly disagree, I think we can stop changing it. cc @qiujian16 

> rather than backup&restore needing to figure out all the permutations of changing on clusters out there.

sorry, I don't quite understand this, Could you give more explanations?

I hope this helps. And any suggestions or other questions are welcome.